Automattic/pocket-casts-android
# Add categories to recommendations screen

| 📘 Project: #524  |
| --- |

## Description
Adds rows for our categories to the onboarding recommendations screen based on @adamzelinski 's [feedback](https://github.com/Automattic/pocket-casts-android/issues/524#issuecomment-1347960181).

I implemented this so that the sections are loaded one at a time. I thought this would be good for users with slow connections so that they didn't have to either wait for all the sections to load or have the sections moving around on them as the network responses came back. Let me know if you think there's a better way to handle this.

## Testing Instructions
1. Open a fresh install of the app so that the onboarding flow is triggered. You may want to also [make this change](https://gist.github.com/mchowning/f60082a45df993dd7c0fffbecb60585b) so that the onboarding flow just goes straight to the recommendations screen. Otherwise, navigate through the flow to the recommendations screen.
2. From the recommendations screen, verify that 6 podcasts are shown for each section, first "Trending" and then one section for each "category" from our Discover feed.
3. Verify that tapping the "More ____" button adds an additional 6 podcasts to that section and that the following event is sent: `recommendations_more_tapped, Properties: {"section":"trending","number_visible":6, ...`.
4. Verfiy that landscape mode also looks good and the "More ___" button works as expected.
5. Change your device's language
6. Observe that the titles of the sections change to the new language
7. Observe that the `section` part of the analytics event that gets sent when you tap one of the "More ____" buttons does _**not**_ get translated (so they remain consistent regardless of the device's language).

## Screenshots or Screencast 

https://user-images.githubusercontent.com/4656348/210627440-ff23f1e3-f1ea-43d6-bd88-db4385f81685.mov

## Checklist
 
#### I have tested any UI changes...
